cli/command_snapshot_delete.go

```diff
@@ -0,0 +1,57 @@
+package cli
+
+import (
+	"context"
+
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/manifest"
+	"github.com/pkg/errors"
+)
+
+var (
+	snapshotDeleteCommand        = snapshotCommands.Command("delete", "Explicitly delete a snapshot by providing a snapshot ID.")
+	snapshotDeleteID             = snapshotDeleteCommand.Arg("id", "Snapshot ID to be deleted").Required().String()
+	snapshotDeleteHostName       = snapshotDeleteCommand.Flag("hostname", "Specify the host name of the snapshot to be deleted").String()
+	snapshotDeleteUserName       = snapshotDeleteCommand.Flag("username", "Specify the user name of the snapshot to be deleted").String()
+	snapshotDeletePath           = snapshotDeleteCommand.Flag("path", "Specify the path of the snapshot to be deleted").String()
+	snapshotDeleteIgnoreHostName = snapshotDeleteCommand.Flag("unsafe-ignore-host", "Override the requirement to specify a host name for the delete to succeed").Bool()
+	snapshotDeleteIgnoreUserName = snapshotDeleteCommand.Flag("unsafe-ignore-user", "Override the requirement to specify a user name for the delete to succeed").Bool()
+	snapshotDeleteIgnorePath     = snapshotDeleteCommand.Flag("unsafe-ignore-path", "Override the requirement to specify a path for the delete to succeed").Bool()
+)
+
+func runDeleteCommand(ctx context.Context, rep *repo.Repository) error {
+	if !*snapshotDeleteIgnoreHostName && *snapshotDeleteHostName == "" {
```

Comment:
Using the helpers mentioned above, then `hostname` and `username` are expected to be set to default values. And even if they are set to empty, then the checks below would fail anyway because they should never be empty in a snapshot manifest.